lib/gtfs/cache_file.ex

```diff
@@ -0,0 +1,113 @@
+defmodule Gtfs.CacheFile do
+  @moduledoc """
+  This module holds the logic for loading and saving cache files
+  in dev and test environments with the goal of decreasing startup
+  time so devs can work more quickly.
+
+  If GTFS data changes, the cache files should be deleted/removed
+  to allowing caching of the changes.
+  """
+  require Logger
+
+  @directory Path.join([File.cwd!(), "priv/gtfs_cache"])
+
+  @doc """
+  The application should use the file to load data for GTFS
+  if the the envs are :dev or :test.
+
+    iex> Mix.env
+    :test
+    iex> CacheFile.should_use_file?
+    true
+  """
+  @spec should_use_file?() :: boolean
+  def should_use_file? do
+    cache_filename() != nil
+  end
+
+  @doc """
+  Attempt to load a cache file.
+
+  This will attempt to load a cache file in dev and test. It will fail
+  if the Mix.env is not dev or test, it will fail if the file does not exist
+  and it will fail if the loaded term is not a map (minimal validation).
+
+  The validation for loading this file must be appended in later code. No
+  such validator currently exists.
+  """
+  @spec load_gtfs() :: {:ok, Gtfs.t()} | {:error, atom}
+  def load_gtfs() do
+    Logger.info(fn -> "Loading gtfs cache from default path" end)
+
+    cache_filename()
+    |> generate_filepath()
+    |> load_gtfs()
+  end
+
+  @spec load_gtfs(String.t()) :: {:ok, Gtfs.t()} | {:error, atom}
+  def load_gtfs(filepath) when is_binary(filepath) do
+    Logger.info(fn -> "Loading gtfs cache from file #{filepath}" end)
+
+    with {:ok, binary_cache} <- File.read(filepath),
+         {:ok, data} when is_map(data) <- binary_to_term(binary_cache) do
+      Logger.info(fn -> "Loaded gtfs cache from file #{filepath}" end)
+      {:ok, data}
+    else
+      _ ->
+        Logger.info(fn -> "Failed to load gtfs cache from file #{filepath}" end)
+        {:error, :cache_not_loaded}
+    end
+  end
+
+  @doc """
```

Comment:
This isn't tested: https://codecov.io/gh/mbta/skate/compare/8c56a4beab5aeac5fb15c6fa060c057137e12124...7bd5f55fe020d2348fc4edc0c767f60c21ca0252/diff#D9-62